lms/djangoapps/support/static/support/jsx/entitlements/data/actions/entitlement.js

```diff
@@ -28,8 +28,73 @@ const fetchEntitlements = username =>
     );
   };
 
+const reissueEntitlementSuccess = entitlement => ({
+  type: entitlementActions.reissue.SUCCESS,
+  entitlement,
+});
+
+const reissueEntitlementFailure = error =>
+  dispatch =>
+    dispatch(displayError('Error Reissuing Entitlement', error));
+
+const reissueEntitlement = ({ entitlement, comments }) =>
+  (dispatch) => {
+    patchEntitlement({
+      uuid: entitlement.uuid,
+      action: 'REISSUE',
+      unenrolledRun: entitlement.enrollmentCourseRun,
+      comments,
+    })
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error(response);
+    })
+    .then(
+      json => dispatch(reissueEntitlementSuccess(camelize(json))),
+      error => dispatch(reissueEntitlementFailure(error)),
```

Comment:
`reissueEntitlementSuccess` just returns a function but it looks like `reissueEntitlementFailure` returns a function that returns a function, right?

Will this cause problems with `dispatch`?

Reply:
Nope, its just a chained redux-thunk action, not necessarily the cleanest, but works.